### PR TITLE
Add types for requestOptions

### PR DESCRIPTION
**Pull request by pedrodurek**
Closes https://github.com/i18next/i18next-http-backend/issues/67

#### Checklist

- [x] only relevant code is changed (make a diff before you submit the PR)
- [x] run tests `npm run test`
- [ ] tests are included
- [ ] documentation is changed or added

### CODE REVIEW

**Comment by adrai** [comment]
released with v1.2.5

File: index.d.ts [hunk 1]
```diff
@@ -74,6 +74,11 @@ interface BackendOptions {
    * interval (useful in server environments)
    */
   reloadInterval?: false | number;
+
+  /**
+   * fetch api request options, can be a function
+   */
+  requestOptions: RequestInit | ((payload: {} | string) => RequestInit);
```
Comment (emsa16) [inline index.d.ts:81]: Shouldn't it be optional?

Comment (adrai) [reply index.d.ts:81]: yes

Comment (adrai) [reply index.d.ts:81]: v1.2.6

**Pull request merged**